addons/website_project/controllers/main.py

```diff
@@ -30,9 +30,19 @@ def extract_data(self, model, values):
         data = super().extract_data(model, values)
         if model.model == 'project.task' and values.get('email_from'):
             partners_list = request.env['mail.thread'].sudo()._mail_find_partner_from_emails([values['email_from']])
-            partner_id = partners_list[0].id if partners_list else False
-            data['record']['partner_id'] = partner_id
+            partner = partners_list[0] if partners_list else self.env['res.partner']
             data['record']['email_from'] = values['email_from']
```

Comment:
We could also keep the other line to force partner_id to False otherwise an existing compute could set it for instance.

```suggestion
            data['record']['partner_id'] = partner.id
            data['record']['email_from'] = values['email_from']
```